chirpstack/src/api/tenant.rs

```diff
@@ -334,6 +334,20 @@ impl TenantService for Tenant {
             )
             .await?;
 
+        let auth_id = request.extensions().get::<AuthID>().unwrap();
+        let id = match auth_id {
+            AuthID::User(id) => id,
+            _ => {
+                return Err(Status::internal("no user id"));
+            }
+        };
+
+        if id == &user_id {
+            return Err(Status::invalid_argument(
+                "you can not delete yourself from the user",
+            ));
+        }
+
```

Comment:
Good one, an user should not be able to delete itself. However, this might not be the correct approach. Note that the `AuthID` holds both the User ID and API key ID. The 

```rust
            _ => {
                return Err(Status::internal("no user id"));
            }
```

Prevents deleting users using the API. Better is using something like:

```rust
if let AuthID::User(id) = auth_id {
     if id == user_id {
        // Return the error
    }
}
```

Same applies for the `users.rs` changes`

Reply:
Thanks, @brocaar 
I'll update PR as per your review point

Reply:
Hi @brocaar 
I've updated this PR.